include/libtorrent/aux_/session_impl.hpp

```diff
@@ -765,7 +766,7 @@ namespace libtorrent
 
 			// a thread pool used for async_write_some calls,
 			// to distribute its cost to multiple threads
-			std::vector<boost::shared_ptr<network_thread_pool> > m_net_thread_pool;
+			std::vector<std::unique_ptr<network_thread_pool>> m_net_thread_pool;
```

Comment:
heh, running across this old code this morning put it pretty high on my list of things to kill :) or at least greatly simplify its implementation